src/Microsoft.Health.Fhir.Liquid.Converter/Json/JsonDataParser.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Models;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Liquid.Converter.Json
+{
+    public class JsonDataParser
+    {
+        public IDictionary<string, object> Parse(string json)
+        {
+            if (string.IsNullOrWhiteSpace(json))
+            {
+                throw new DataParseException(FhirConverterErrorCode.NullOrWhiteSpaceInput, Resources.NullOrWhiteSpaceInput);
+            }
+
+            try
+            {
+                var data = JsonConvert.DeserializeObject<object>(json, new JsonSerializerSettings { ContractResolver = new DictionaryContractResolver() });
```

Comment:
>DeserializeObject [](start = 39, length = 17)

how about using JObject.Parse?

Reply:
JObject.Parse does not accept `ContractResolver` as input. Here we need `DictionaryContractResolver` to parse input data to `IDictionary<string, object>` instead of `JObject` /`JArray`, which is not accepted by dotliquid.

Reply:
`DeserializeObject` is replaced with `JToken.Parse` and `JTokenExtensions` in the update.